### DISCUSSION TITLE
What is the recommended way to achieve Request Logging in App Router?

### DISCUSSION THREAD

**Discussion by vishwa795** [vercel/next.js#65176] [App Router]
For most major applications, request logging becomes a pretty important thing to have. After digging around for almost a day, all i could come up with was a middleware, which can log request details. I haven't been able to find any source which would help in logging things like response time, response status etc. 

How can i achieve this?

Ideally app router should also have the ability to create a custom server. This was a simple problem to solve in earlier next versions, with custom servers. But now with next 14, there's no way we can add logging.

Furthermore, the logs due to `logging.fetches.fullUrl` are helpful, but only work on development server. There should be a config here to display the logs in production server as well.

**Comment by richg0ld** [comment]
I am also experiencing the same issue. I would like the `logging.fetches.fullUrl` option to be functional in production environments as well.

**Comment by yiichou** [comment]
I would like the `logging.fetches.fullUrl` to be available in the production environment as well.

It sounds unreliable for a backend service to work without request logs.
